source/Plugins/ExpressionParser/Clang/ClangHost.cpp

```diff
@@ -53,9 +36,36 @@ static bool VerifyClangPath(const llvm::Twine &clang_path) {
   return false;
 }
 
-bool lldb_private::ComputeClangDirectory(FileSpec &lldb_shlib_spec,
+///
+/// This will compute the clang resource directory assuming that clang was
+/// installed with the same prefix as lldb.
+///
+static bool DefaultComputeClangResourceDirectory(FileSpec &lldb_shlib_spec,
                                          FileSpec &file_spec, bool verify) {
   std::string raw_path = lldb_shlib_spec.GetPath();
+  llvm::StringRef parent_dir = llvm::sys::path::parent_path(raw_path);
+
+  llvm::SmallString<256> clang_dir(parent_dir);
+  llvm::SmallString<32> relative_path;
+  llvm::sys::path::append(relative_path, "lib", "lldb", "clang");
```

Comment:
@xiaobai This is now different from llvm.org, where this line reads:
```
  llvm::sys::path::append(relative_path,
                          llvm::Twine("lib") + CLANG_LIBDIR_SUFFIX, "clang",
                          CLANG_VERSION_STRING);
```

Why is that? And could you please make sure to bring both versions back in sync?

Reply:
From what I understand, swift-lldb relies on the clang resource copied from swift, which is placed in `$install_location/lib/lldb/clang`, whereas llvm.org's lldb relies on the clang resource dir located at `$install_location/lib/clang/$clang_version`. I'm not sure how I'd synchronize llvm.org and swift-lldb right now. Maybe a cmake variable that fills out Config.h?

Reply:
I think you should be able to do something similar to what the macOS implementation does and just have a list of locations that are searched one after another. You can probably even reuse some of the code in the macOS implementation. If you think that that is too inefficient then we should detect the build type (e.g., by having CMake generate a macro) and make the code depend on that. Depending on the repository/branch is not a good idea, because some people *will* be building the swift-lldb compiler without swift.

Reply:
Ah yeah looking at what the MacOS implementation does, that sounds like the best option. I should have some time in the next few days to fix that up. Thanks for pointing this out

Reply:
Thanks! These kinds of differences between upstream and swift-lldb often cause extremely difficult to debug problems, especially when related to search paths, so this will pay off quite a lot.

Reply:
@adrian-prantl I opened #1411 to cherry-pick the change to stable.